src/cosl/coordinated_workers/nginx.py

```diff
@@ -1,12 +1,164 @@
 # Copyright 2023 Canonical
 # See LICENSE file for licensing details.
 
-"""Workload manager for Nginx. Used by the coordinator to load-balance and group the workers."""
+r"""## Overview.
+
+This module provides a set of utilities for generating Nginx configurations and managing Nginx workloads.
+Used by the coordinator to load-balance and group the workers.
+
+- `NginxLocationModifier`: An enum representing valid Nginx `location` block modifiers (e.g., `=`, `~`, `^~`).
+Should be used to populate an `NginxLocationConfig` object.
+
+- `NginxLocationConfig`: A class that defines a single Nginx `location` block, including path matching, modifier, custom headers, .etc.
+
+- `NginxUpstream`: A class that describes an Nginx `upstream` block — specifying an upstream `name`, `port`, and `worker_role` used to resolve backend endpoints.
+
+- `NginxConfig`: A class that builds a full Nginx configuration to be used by the coordinator to load-balance traffic across the workers.
+
+- `Nginx`: A helper class for managing the Nginx container workload (e.g., pebble service lifecycle, config reloads).
+
+- `NginxPrometheusExporter`: A helper class for managing the Nginx Prometheus exporter container workload (e.g., pebble service lifecycle, config reloads).
+
+- `is_ipv6_enabled()`: A utility function to check whether IPv6 is enabled on the container's network interfaces.
+
+## Usage
+### Nginx Config Generation
+
+To generate an Nginx configuration for a charm, instantiate the `NginxConfig` class with the required inputs:
+
+1. `server_name`: The name of the server (e.g. charm fqdn), which is used to identify the server in Nginx configurations.
+2. `upstream_configs`: List of `NginxUpstream` used to generate Nginx `upstream` blocks.
+3. `server_ports_to_locations`: Mapping from server ports to a list of `NginxLocationConfig`.
+
+#### Use `NginxConfig` in the context of the shared `Coordinator` object
+
+A coordinator charm may instantiate the `NginxConfig` in its constructor as follows:
+    from cosl.coordinated_workers.nginx import NginxConfig, NginxUpstream, NginxLocationConfig
+    ...
+
+    class CoordinatorCharm(CharmBase):
+        ...
+        def __init__(self, *args):
+            super().__init__(*args)
+            ...
+            self.coordinator = Coordinator(
+                ...
+                nginx_config=NginxConfig(
+                    server_name=self.hostname,
+                    upstream_configs=self._nginx_upstreams(),
+                    server_ports_to_locations=self._server_ports_to_locations(),
+                )
+            )
+        ...
+        @property
+        def hostname(self) -> str:
+            return socket.getfqdn()
+
+        @property
+        def _nginx_locations(self) -> List[NginxLocationConfig]:
+            return [
+                NginxLocationConfig(path="/loki/api/v1/rules", backend="backend",modifier=NginxLocationModifier("=")),
```

Comment:
this is how I'd use an enum
```suggestion
                NginxLocationConfig(path="/loki/api/v1/rules", backend="backend",modifier=NginxLocationModifier.EXACT),
```

Which makes me think, if it's better to use Typing.Literal instead of this humongously-named enum if all we're really doing it for is type hinting the input.